frontend/src/main/java/com/google/tripmeout/frontend/storage/PlaceVisitStorage.java

```diff
@@ -24,40 +28,46 @@
    * @param tripId the id to match PlaceVisitModel object's placeId field on
    * @param placeId the id to match PlaceVisitModel object's placeId field on
    *
-   * @throws a PlaceVisitNotFound exception if there is no PlaceVisitModel 
-   *     object in storage with the given tripId and placeId 
+   * @throws a PlaceVisitNotFound exception if there is no PlaceVisitModel
+   *     object in storage with the given tripId and placeId
    */
-  void removePlaceVisit(String tripId, String placeId);
+  void removePlaceVisit(String tripId, String placeId) throws PlaceVisitNotFoundException;
 
   /**
    * returns the PlaceVisitModel whose tripId and placeId match the given tripId
    * and placeId, respectively
    *
    * @param tripId the id to match PlaceVisitModel object's placeId field on
    * @param placeId the id to match PlaceVisitModel object's placeId field on
-   *
-   * @throws a PlaceVisitNotFound exception if there is no PlaceVisitModel 
-   *     object in storage with the given tripId and placeId 
    */
-  PlaceVisitModel getPlaceVisit(String tripId, String placeId);
+  Optional<PlaceVisitModel> getPlaceVisit(String tripId, String placeId);
 
   /**
-   * updates the userMark parameter of the PlaceVisitModel object whose tripId 
+   * updates the userMark parameter of the PlaceVisitModel object whose tripId
    * and placeId match the given tripId and placeId, respectively
+   * return true if PlaceVisitModel object with the given tripId and placeId
+   * exist in storage
    *
    * @param tripId the id to match PlaceVisitModel object's tripId field on
    * @param placeId the id to match PlaceVisitModel object's placeId field on
-   *
-   * @throws a PlaceVisitNotFound exception if there is no PlaceVisitModel 
-   *     object in storage with the given tripId and placeId 
    */
-  void changePlaceVisitStatus(String tripId, String placeId, String newStatus);
+  boolean changePlaceVisitStatus(String tripId, String placeId, String newStatus);
```

Comment:
Not blocking this, but why change this to a boolean from exception-throwing? Do we expect calls to `changePlaceVisitStatus` on trip/place combos that don't exist to be common? Why would that happen? Note: it's easy to forget to check a boolean return value like this (unless you enable additional static analysis tools, which I don't believe we have right now), while it's impossible to ignore a thrown exception.

The correct choice depends on the use case(s) that you have in mind.

Reply:
I was imagining something where we can call this function regardless of whether the user is changing the status from a previously selected status or selecting a status for the first time and that PlaceVisit objects wouldn't be stored until after the user has marked the place with a status